### PR TITLE
Improve PDM microphone reliability

### PR DESCRIPTION
**Pull request by facchinm**

### CODE REVIEW

File: libraries/PDM/src/PDM.cpp [hunk 1]
```diff
@@ -48,10 +48,15 @@ int PDMClass::begin(int channels, long sampleRate)
 {
   _channels = channels;
 
+  NRF_CLOCK->EVENTS_HFCLKSTARTED = 0;
+  NRF_CLOCK->TASKS_HFCLKSTART    = 1;
+  while (NRF_CLOCK->EVENTS_HFCLKSTARTED == 0) { }
```
Comment (sandeepmistry) [inline libraries/PDM/src/PDM.cpp]: Should this be disabled in `.end()`?

Comment (facchinm) [reply libraries/PDM/src/PDM.cpp]: I thought about this and it might not be clever (since BLE could be in use and would stop working on `PDM.end()` ).

Comment (sandeepmistry) [reply libraries/PDM/src/PDM.cpp]: ok, so then should we check if it's already enabled before "restarting" it. I would also add a small comment to capture why end() doesn't disable to be kind to our future selves :)

Comment (facchinm) [reply libraries/PDM/src/PDM.cpp]: Agree on the comment for our future selves :wink:

**Pull request merged**